src/coreclr/vm/corelib.cpp

```diff
@@ -35,6 +35,7 @@
 #include "commodule.h"
 #include "marshalnative.h"
 #include "nativelibrarynative.h"
+#include "nativememorynative.h"
```

Comment:
I don't think that's easy given that we don't know what `libc` we are compiling against or will be available at runtime.

Trying to bind against `msvcrt` or `libc` has always been a major pain.

Reply:
It will take care of the Mono (and NativeAOT) implementation that would have to be duplicated otherwise.

Reply:
I think we should duplicate it, or have some special `DllImport` helper for `libc`.

Otherwise we need to bind against one of:
* `msvcrt` on most Windows
* `libc.so.6` on most Linux (Just `libc` doesn't always resolve on some platforms, and is something I had to deal with in my `mimalloc` port)
* `libSystem` on MacOS

and deal with the possibility that the exports are in one of the "other" libc libraries or not exported under the "simple" name, etc

Reply:
I'm also not quite sure how this will impact iOS/Android and whether `libc`/`libSystem` is the "correct" place to get these

Reply:
On Windows, it should PInvoke ucrtbase.dll that is the proper public C library implementation. msvcrt is legacy and it was never really public API.

On non-Windows, it should PInvoke the System.Native shim just like the existing `Marshal.Alloc/Free` APIs. Note that you can  reuse the existing shims for malloc/realloc/free, just need to add new ones for the more niche functionality.

Reply:
> On Windows, it should PInvoke ucrtbase.dll that is the proper public C library implementation. msvcrt is legacy and it was never really public API.

Is `ucrtbase.dll` available on Win7 (I think we still support Win7 at least)?

Is `ucrtbase` considered "public API" and are there any concerns with it being changed in the future?

Reply:
> Is ucrtbase.dll available on Win7 (I think we still support Win7 at least)?

Yes. coreclr.dll depends on ucrtbase.dll.  We have it as [prereq](https://docs.microsoft.com/en-us/dotnet/core/install/windows?tabs=net50#additional-deps) or bundle for self-contained .

Reply:
Thanks for the confirmation. I'll update this.

Reply:
> Is ucrtbase considered "public API" 

Technically, the public API is exported via `api-ms-win-crt-*.dll`. If you would like to do the right thing (tm), you can DllImport it via the API sets.

Personally, I would be fine with just using ucrtbase.dll for simplicity too. There are so many places where uctrbase.dll is hardcoded so it is very unlikely to ever change, and we have been largely ignorant of API sets.